msm/skill_state.py

```diff
@@ -1,16 +1,33 @@
 """Functions related to manipulating the skills.json file."""
 import json
+import os
 from logging import getLogger
-from os.path import expanduser, isfile, dirname
+from os.path import isfile, dirname, join
 from os import makedirs
+import os
+from xdg import BaseDirectory
 
 LOG = getLogger(__name__)
-SKILL_STATE_PATH = '~/.mycroft/skills.json'
+
+
+def get_state_path():
+    """Get complete path for skill state file.
+
+    Returns:
+        (str) path to skills.json
+    """
+    return join(BaseDirectory.save_data_path('mycroft'), 'skills.json')
+
+
+# Make sure we migrate the installed skills file from the old non-XDG location
+old_skill_state_path = '~/.mycroft/skills.json'
+if isfile(old_skill_state_path):
+    os.rename(old_skill_state_path, get_state_path())
```

Comment:
I don't think it's an issue but it's safer to use shutil.move in case of moving across partition boundaries.